docs/client-sdk/javascript/tutorials/build-an-xmtp-hello-world-app.md

```diff
@@ -0,0 +1,576 @@
+
+2. Sign to enable their XMTP identity the first and any subsequent time they start a new messaging session (like providing an account password).
+
+To learn more about creating and enabling an XMTP identity, see [Sign to send and receive messages using apps built with XMTP](https://xmtp.org/docs/dev-concepts/account-signatures).
+
+
+**Build the Connect to XMTP button:**
+
+
+1. Install the [XMTP client SDK for JavaScript](https://github.com/xmtp/xmtp-js).
+
+   The SDK provides an XMTP message API client that enables your "GM Bot" app to communicate with the XMTP network. In the `my-app` directory, run:
+
+   ```bash
+   npm i --save @xmtp/xmtp-js
+   ```
+
+   This command adds the `@xmtp/xmtp-js` dependency to your `package.json`.
+
+
+2. Before we connect XMTP with our button, let's first create some conditions that ensure that the app shows the  **Connect to XMTP** button only if a wallet is connected.
+   ```tsx title="App.tsx" showLineNumbers
+   import React from 'react';
+   import './App.css';
+   import {
+     EthereumClient,
+     modalConnectors,
+     walletConnectProvider,
+   } from "@web3modal/ethereum";
+   import { Web3Modal, Web3Button } from "@web3modal/react";
+   import { configureChains, createClient,} from "wagmi";
+   import { mainnet, polygon } from "wagmi/chains";
+
+   const chains = [ mainnet, polygon];
+   // Wagmi client
+   const { provider } = configureChains(chains, [
+     walletConnectProvider({ projectId: "<YOUR_PROJECT_ID>" }),
+   ]);
+   const wagmiClient = createClient({
+     autoConnect: true,
+     connectors: modalConnectors({ appName: "web3Modal", chains }),
+     provider,
+   });
+   // Web3Modal Ethereum Client
+   const ethereumClient = new EthereumClient(wagmiClient, chains);
+
+   function App() {
+     // highlight-next-line
+     const { isConnected } = useAccount()
+
+     return (
+       <div className="App">
+         <Web3Button/>
+         <Web3Modal
+           projectId="<YOUR_PROJECT_ID>"
+           ethereumClient={ethereumClient}
+         />
+         // highlight-next-line
+        {isConnected && <button onClick={initXmtp}>Connect to XMTP</button>}
+       </div>
+     );
+   }
+
+   export default App;
+   ```
+   So now, the **Connect to XMTP** button should only show when your wallet is connected.
+   
+3. Now, let's add an onClick handler when users click the button. This will initialize XMTP with your app.
+   ```tsx title="App.tsx" showLineNumbers
+   import React from 'react';
+   import './App.css';
+   import {
+     EthereumClient,
+     modalConnectors,
+     walletConnectProvider,
+   } from "@web3modal/ethereum";
+   import { Web3Modal, Web3Button } from "@web3modal/react";
+   import { configureChains, createClient,} from "wagmi";
+   import { mainnet, polygon } from "wagmi/chains";
+   // highlight-start
+   import { fetchSigner } from '@wagmi/core'
+   import { Client } from '@xmtp/xmtp-js'
```

Comment:
```suggestion
   import { fetchSigner } from '@wagmi/core';
   import { Client } from '@xmtp/xmtp-js';
```

I'm definitely not a dev, but I see these `;` in the subsequent code samples - so thought it would be good to add them here too.

Reply:
ahh yeah, missed those. Prettier doesnt catch these on markdown files. Thanks!